### PR TITLE
Allow to dereference Follow object for ActivityPub

### LINKED ISSUE #5770: id of Follow object is not dereferencable
Currently, it uses a fragment for `id` of `Follow` object, but, strictly speaking, that is a violation of the specification.

* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

### PR DESCRIPTION
**Pull request by akihikodaki**
Resolves #5770.

### CODE REVIEW

**Comment by akihikodaki** [comment]
7a7604908ffa9cf9377a33a22b5bc7ff2002b813 is a change suggested at https://github.com/tootsuite/mastodon/pull/5649#pullrequestreview-77766123. 7a7604908ffa9cf9377a33a22b5bc7ff2002b813 must be merged before #5649.

**Review by Gargron — APPROVED** [review]

File: app/serializers/activitypub/follow_serializer.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 class ActivityPub::FollowSerializer < ActiveModel::Serializer
-  attributes :id, :type, :actor
+  attributes :type, :actor
+  attribute :id, if: -> { object.respond_to(:object_type) }
```
Comment (Gargron) [inline app/serializers/activitypub/follow_serializer.rb]: `respond_to?` (missing question mark)

I would also prefer using `if: :symbol` with a method definition as it's easier to read. E.g. `if: :supports_dereferencing?`

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by akihikodaki** [comment]
55ecc966a07fa1b523843539c5f565d9454e7e5a resolves https://github.com/tootsuite/mastodon/pull/5772#pullrequestreview-86140318.

File: app/serializers/activitypub/follow_serializer.rb [hunk 2]
```diff
@@ -19,4 +20,10 @@ def actor
   def virtual_object
     ActivityPub::TagManager.instance.uri_for(object.target_account)
   end
+
+  private
```
Comment (Gargron) [inline app/serializers/activitypub/follow_serializer.rb]: Unfortunately it can't be private in a serializer (see build failure)

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by akihikodaki** [comment]
9818b2f0cc3d6281c944084cfa08158df91560bd resolves https://github.com/tootsuite/mastodon/pull/5772#pullrequestreview-86292893.

**Review by Gargron — APPROVED** [review]

**Pull request merged**